crates/oxc_semantic/src/builder.rs

```diff
@@ -151,7 +151,9 @@ impl<'a> SemanticBuilder<'a> {
     }
 
     pub fn build(mut self, program: &Program<'a>) -> SemanticBuilderReturn<'a> {
-        if !self.source_type.is_typescript_definition() {
+        if self.source_type.is_typescript_definition() {
+            self.scope.add_scope(None, ScopeFlags::Top);
+        } else {
             self.visit_program(program);
```

Comment:
probably not the best solution - happy to change either `root_unresolved_references` to return an option or an empty FxHashMap if the scope has not been visited.

But doing it that way felt like extra overhead that is almost never needed